File: 485-max-consecutive-ones/485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        
        if (nums.length ==1) return nums[0];
        
        int cnt =0 ; 
        int maxv =0 ; 
        for (int i=0; i<nums.length; i++){
            
            if(nums[i]==1){
                cnt ++;
                maxv = Math.max(cnt , maxv);
                
            }
            if( i>=1  && nums[i] ==0 && nums[i-1] ==1){
                cnt =0; 
            }
            
        }
        return maxv;
        
        
    }
}

File: 485-max-consecutive-ones/485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        
        if len(nums)==1:
            
            return nums[0]
        
        
        cnt = 0
        maxv = 0 # no -sys.maxsize
        for i in range(len(nums)):

            if nums[i] ==1:
                cnt +=1
                maxv = max(cnt, maxv)

            if nums[i] ==0 and nums[i-1] ==1: # there may be an edge case, here! 
                cnt =0 
        return maxv 